### ISSUE TITLE
[Bug]: `act` from `react-dom/test-utils` is deprecated

### ISSUE THREAD

**Issue by MathiasSM** [cloudscape-design/components#2471]
### Package version

v3.0.1282

### React version

v18.3.1

### Description

All over the test utils in cloudscape components we're using act react-dom/test-utils, which is generating console.errors and stack traces (for each use) like the following:

```
Warning: `ReactDOMTestUtils.act` is deprecated in favor of `React.act`. Import `act` from `react` instead of `react-dom/test-utils`. See https://react.dev/warnings/react-dom-test-utils for more info.
```

Expectation is for this library to use the new path (even if just to avoid a lot of noise in the testing logs of users).

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/cloudscape-design/components/blob/main/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/cloudscape-design/components/issues) for duplicate problems

**Comment by abdhalees** [comment]
Hello MathiasSM@,

Since this warning is shown by react as preparation for their React 19 release, I've removed the `bug` label as warnings are not bugs.

We agree, it's something we need to address. do you want to contribute this enhancement?

**Comment by MathiasSM** [comment]
While the `test-utils-core` package was fixed, the `test-utils` within specific components (this package) was not. 

- There are around 10 or so ocurrences in non-test code (which should use either act exported from `test-utils-core`, or repeat the logic here, checking if `React.act` exists), and 
- there are a few more within `__test__` directories (which should use `@testing-library/react`'s `act`, which does the same check plus a few more things.

I tried building both components and test-utils, but failed miserably. I feel the setup is assuming default or global configuration of typescript libs, as it complains about it giving many errors related to missing base types.

As a note, you're also using `react-dom/test-utils`'s `Simulate`, which is also deprecated. The recommended is to use `@testing-library/react` fireEvent (https://react.dev/warnings/react-dom-test-utils) but that requires you adding testing-library to production/normal dependencies, and I don't think I'm the right guy to evaluate whether that's ok (or how to go about it, like using peer deps, or straight up documenting that using testing-library is _the way_ to test and not just a recommendation(?))
